src/Tribe/Ignored_Events.php

```diff
@@ -0,0 +1,526 @@
+		 * @return array $views After adding the new Link
+		 */
+		public function add_ignored_view( $views = array() ) {
+			// Get the Old ones
+			$counter = $this->count_legacy_deleted_posts();
+			if ( 0 >= $counter ) {
+				return $views;
+			}
+
+			// This Will prevent having the Ignored twice on the Edit Page
+			if ( $this->has_ignored_posts( false ) ) {
+				return $views;
+			}
+
+			$args = array(
+				'post_type' => Tribe__Events__Main::POSTTYPE,
+				'ignored_events' => 1,
+			);
+			$url = add_query_arg( $args, 'edit.php' );
+
+			$views['import-deleted'] = '<a class="' . ( isset( $_GET['ignored_events'] ) ? 'current' : '' ) . '" href="' . esc_url( $url ) . '">' . esc_html__( 'Ignored', 'the-events-calendar' );
+			$views['import-deleted'] .= sprintf( ' <span class="count">(%d)</span></a>', $counter );
+
+			return $views;
+		}
+
+		/**
+		 * Count legacy "ignored" posts
+		 *
+		 * @return int
+		 */
+		public function count_legacy_deleted_posts() {
+			// Fetch any posts on the `deleted_event` CPT
+			$counter = wp_count_posts( self::$legacy_deleted_post );
+
+			// Check if we have trash status set
+			if ( ! isset( $counter->trash ) ) {
+				return 0;
+			}
+
+			// Return it been a Int
+			return absint( $counter->trash );
+		}
+
+		/**
+		 * Check if there are any legacy posts
+		 *
+		 * @return boolean
+		 */
+		public function has_legacy_deleted_posts() {
+			return $this->count_legacy_deleted_posts() > 0;
+		}
+
+		/**
+		 * Check if there are any ignored posts
+		 *
+		 * @param  boolean $check_legacy If the method should also check legacy CPTs
+		 * @return boolean
+		 */
+		public function has_ignored_posts( $check_legacy = true ) {
+			$has_legacy = $this->has_legacy_deleted_posts();
+			if ( true === $check_legacy && true === $has_legacy ) {
+				return true;
+			}
+
+			$query = new WP_Query( array(
+				'fields' => 'ids',
+				'post_type' => Tribe__Events__Main::POSTTYPE,
+				'post_status' => self::$ignored_status,
+			) );
+
+			return $query->have_posts();
+		}
+
+		/**
+		 * Changes the event to the correct Post Status
+		 *
+		 * @param  int|WP_Post       $event Which event try to convert
+		 * @return bool|int|WP_Error
+		 */
+		public function ignore_event( $event ) {
+			$event = get_post( $event );
+
+			if ( ! $event instanceof WP_Post ) {
+				return false;
+			}
+
+			// Update only what we need
+			$arguments = array(
+				'ID' => $event->ID,
+				'post_status' => self::$ignored_status,
+			);
+
+			// Try to update back to the Event CPT
+			return wp_update_post( $arguments );
+		}
+
+		/**
+		 * Restore Event
+		 *
+		 * @param  int|WP_Post       $event Which event try to convert
+		 * @return bool|int|WP_Error
+		 */
+		public function restore_event( $event ) {
+			$event = get_post( $event );
+
+			if ( ! $event instanceof WP_Post ) {
+				return false;
+			}
+
+			// Update only what we need
+			$arguments = array(
+				'ID' => $event->ID,
+				'post_status' => 'publish',
```

Comment:
Just a thought: if you trash a WP post (that had a status of 'private') then restore it, it is set back to private rather than being published to the world. Should we follow suit here?